ocaml/xapi/xapi_vm_helpers.ml

```diff
@@ -392,6 +392,15 @@ let assert_usbs_available ~__context ~self ~host =
         (Ref.string_of host)
       ]))
     )
+let assert_netsriov_available ~__context ~self ~host = 
+  Xapi_network_sriov_helpers.get_sriov_networks_from_vm ~__context ~vm:self |>
```

Comment:
Also, I don't think that this takes into account the case of multiple VIFs on the same SR-IOV network for the same VM. It would just check for 1 VF multiple, rather than requiring multiple VF together. So the pipeline above needs to be a little smarter.